### PR TITLE
[Breaking] refactor(eslint-mc-app): simplify and improve usage of eslint config

### PR DESCRIPTION
**Pull request by emmenko**
Our ESLint config hasn't been updated in a while. Furthermore, in this repo we were not using the mc-app config but a "custom" one specific for typescript.

Let's change that and use our own config here as well, with support for TypeScript out of the box.

So what changed here:
* I decided to drop the airbnb base config and use instead the react-app config. The react-app config comes with some good defaults (also regarding TypeScript) and it's anyway what we look for in terms of tooling (react-scripts, babel, etc). So it's more aligned on that front as well.
* No more many required peer dependencies to be installed. I think we should make it as simple as possible for users to set up their dev environment, and having to install >10 peer dependencies does not help much. Even though ESLint recommends to ship a sharable config with peer dependencies, I think the usability is still better if we only require `eslint` as a peer dependency.
  > Note that all the dependencies are defined with a caret `^` version range, which gives a bit of flexibility to the consumers in case of version conflicts.
* Cleaned up some of the rules.
* Included some additional TypeScript config that we were using before anyway (extending what is already specified in react-app config).

### CODE REVIEW

**Comment by tdeekens** [comment]
Nice. Let's finish internal refactors by earlier releases before loading teams with more things I'd suggest. Can you anticipate how many changes this would cause and how many are auto-fixable?

**Comment by emmenko** [comment]
> Can you anticipate how many changes this would cause and how many are auto-fixable?

I tested this in our internal repo. It's all good, and it actually found a bunch of legit errors! 🤓

File: packages/eslint-config-mc-app/README.md [hunk 1]
```diff
@@ -4,7 +4,7 @@
   <a href="https://www.npmjs.com/package/@commercetools-frontend/eslint-config-mc-app"><img src="https://badgen.net/npm/v/@commercetools-frontend/eslint-config-mc-app" alt="Latest release (latest dist-tag)" /></a> <a href="https://www.npmjs.com/package/@commercetools-frontend/eslint-config-mc-app"><img src="https://badgen.net/npm/v/@commercetools-frontend/eslint-config-mc-app/next" alt="Latest release (next dist-tag)" /></a> <a href="https://bundlephobia.com/result?p=@commercetools-frontend/eslint-config-mc-app"><img src="https://badgen.net/bundlephobia/minzip/@commercetools-frontend/eslint-config-mc-app" alt="Minified + GZipped size" /></a> <a href="https://github.com/commercetools/merchant-center-application-kit/blob/master/LICENSE"><img src="https://badgen.net/github/license/commercetools/merchant-center-application-kit" alt="GitHub license" /></a>
 </p>
 
-ESLint config used by a MC application.
+ESLint config used by a (Merchant Center) Custom Application.
```
Comment (adnasa) [inline packages/eslint-config-mc-app/README.md:7]: We don't need `(Merchant Center)`

File: website-components-playground/src/pages/form-dialog.js [hunk 2]
```diff
@@ -3,6 +3,7 @@ import { Formik } from 'formik';
 import { FormDialog } from '@commercetools-frontend/application-components';
 import Spacings from '@commercetools-uikit/spacings';
 import TextField from '@commercetools-uikit/text-field';
+import TextInput from '@commercetools-uikit/text-input';
```
Comment (adnasa) [inline website-components-playground/src/pages/form-dialog.js:6]: how is this related to the eslint udpate? 🤔 
or was this something you just discovered on the way?

**Review by adnasa — APPROVED** [review]

**Comment by emmenko** [comment]
@tdeekens @adnasa I think we're good to go now with this (major) release.

We can also include a bunch more breaking changes, for example "Remove deprecated hostnames" (https://github.com/commercetools/merchant-center-application-kit/issues/1829).

**Review by tdeekens — APPROVED** [review]
Nice. Thanks for iterating on this!
